src/essence/Tools/Isochrone/IsochroneTool_Query.js

```diff
@@ -0,0 +1,153 @@
+import Map_ from "../../Basics/Map_/Map_";
+import F_ from "../../Basics/Formulae_/Formulae_";
+import L_ from "../../Basics/Layers_/Layers_";
+
+const MAX_WORKERS = 8;
+
+const QueryManager = {
+    cache: {},
+    //TODO: queue jobs here
+    //(so e.g. two isochrones will load in sequence w/o slowing each other down)
+    jobQueue: [],
+    jobInProgress: false,
+    numWorkers: 0,
+    currentJob: null,
+
+    //TODO: expand these to maintain a max cache size
```

Comment:
how large can a cache grow to?

Reply:
theoretically, the size of all layers used by models, which could be single-digit GB. practically, no one will run isochrones at every point on the map at every available resolution, but I'd still guess you could pretty easily get it up to 2-digit MB + overhead

Reply:
You could use a limited size queue for the cache to somewhat limit it